src/components/Company/PaySchedule/PaySchedule.tsx

```diff
@@ -0,0 +1,361 @@
+})
+
+export type PayScheduleInputs = v.InferInput<typeof PayScheduleSchema>
+export type PayScheduleOutputs = v.InferOutput<typeof PayScheduleSchema>
+
+export type PayScheduleDefaultValues = RequireAtLeastOne<
+  Partial<
+    Pick<
+      Schemas['Pay-Schedule'],
+      | 'anchor_pay_date'
+      | 'anchor_end_of_pay_period'
+      | 'day_1'
+      | 'day_2'
+      | 'custom_name'
+      | 'auto_pilot'
+    >
+  > & {
+    frequency: 'Every week' | 'Every other week' | 'Twice per month' | 'Monthly'
+  }
+>
+
+const [usePaySchedule, PayScheduleProvider] =
+  createCompoundContext<PayScheduleContextType>('PayScheduleContext')
+export { usePaySchedule }
+
+export function useGetPaySchedulePreview(
+  company_id: string,
+  params: Operations['get-v1-companies-company_id-pay_schedules-preview']['parameters']['query'],
+  enabled = false,
+) {
+  const { GustoClient: client } = useGustoApi()
+  return useQuery({
+    queryKey: ['companies', company_id, 'pay_schedules', 'preview', params],
+    queryFn: async () => {
+      try {
+        return await client.getPaySchedulePreview(company_id, params)
+      } catch (err) {
+        if (err instanceof ApiError) {
+          throw new Error(
+            JSON.stringify({
+              message: err.message,
+              status: err.statusCode,
+              payload: err.errorList,
+            }),
+          )
+        }
+      }
+    },
+    enabled,
+    retry: 0, // Disable retries
+  })
+}
+
+interface PayScheduleProps extends CommonComponentInterface {
+  companyId: string
+  defaultValues?: PayScheduleDefaultValues
+}
+
+export const PaySchedule = ({
+  companyId,
+  defaultValues,
+  ...props
+}: PayScheduleProps & BaseComponentInterface) => {
+  useI18n('Company.PaySchedule')
+  return (
+    <BaseComponent {...props}>
+      <Root companyId={companyId} defaultValues={defaultValues}>
+        {props.children}
+      </Root>
+    </BaseComponent>
+  )
+}
+
+const Root = ({ companyId, children, defaultValues }: PayScheduleProps) => {
+  const { baseSubmitHandler, onEvent } = useBase()
+  const [mode, setMode] = useState<MODE>('LIST_PAY_SCHEDULES')
+  const [currentPaySchedule, setCurrentPaySchedule] = useState<
+    (Schemas['Pay-Schedule'] & { version?: string }) | null
+  >(null)
+  const transformedDefaultValues: PayScheduleInputs = useMemo(() => {
+    return {
+      frequency: defaultValues?.frequency ?? 'Every week',
+      anchor_pay_date: defaultValues?.anchor_pay_date
+        ? parseDate(defaultValues.anchor_pay_date)
+        : undefined,
+      anchor_end_of_pay_period: defaultValues?.anchor_end_of_pay_period
+        ? parseDate(defaultValues.anchor_end_of_pay_period)
+        : undefined,
+      day_1: defaultValues?.day_1 ?? undefined,
+      day_2: defaultValues?.day_2 ?? undefined,
+      custom_name: defaultValues?.custom_name ?? '',
+      auto_pilot: defaultValues?.auto_pilot?.toString() ?? '',
+      custom_twice_per_month: 'false',
+    }
+  }, [defaultValues])
```

Comment:
Nit/question: is it worth memoizing this if default values is an object and will just get compared with strict equality in the dep array?

Reply:
Agreed

Reply:
Resolved